addons/mail/models/mail_channel.py

```diff
@@ -311,12 +311,13 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
             members_to_create = []
             if channel.public == 'groups':
                 invalid_partners = partners.filtered(lambda partner: channel.group_public_id not in partner.user_ids.groups_id)
-                if invalid_partners:
+                if invalid_partners or guests:
+                    partners = invalid_partners or guests
                     raise UserError(_(
                         'Channel "%(channel_name)s" only accepts members of group "%(group_name)s". Forbidden for: %(partner_names)s',
                         channel_name=channel.name,
                         group_name=channel.group_public_id.name,
-                        partner_names=', '.join(partner.name for partner in invalid_partners)
+                        partner_names=', '.join(partner.name for partner in partners)
```

Comment:
You may want to separate the treatment of guests from partners. Both having the same `name` field is a coincidence, I don't think we want to rely on that.